comment_scan/comment_scanner.rb

```diff
@@ -0,0 +1,274 @@
+require './db'
+require_relative 'message_collection'
+
+class CommentScanner
+    attr_reader :seclient, :chatter, :time_to_scan
+
+    def initialize(seclient, chatter, post_all_comments, ignore_users, perspective_key: '', perspective_log: Logger.new('/dev/null'))
+        @seclient = seclient
+        @chatter = chatter
+        @post_all_comments = post_all_comments
+        @ignore_users = ignore_users
+        @perspective_key = perspective_key
+        @perspective_log = perspective_log
+
+        @time_to_scan = 60
+        @latest_comment_date = @seclient.latest_comment_date.to_i+1 unless @seclient.latest_comment_date.nil?
+    end
+
+    def scan_new_comments
+        new_comments = @seclient.comments_after_date(@latest_comment_date)
+        @latest_comment_date = new_comments[0].json["creation_date"].to_i+1 if new_comments.any? && !new_comments[0].nil?
+        scan_se_comments([new_comments])
+    end
+
+    def tick
+        @time_to_scan -= 1
+        if @time_to_scan <= 0
+            @time_to_scan = 60
+            return false
+        end
+        return true
```

Comment:
I kinda like this as a one liner better, although you're welcome to leave it if you prefer (`!(@time_to_scan = 60 if (@time_to_scan -=1) <= 0)`) ... actually... that one liner is a bit hacky. Maybe a two liner `@time_to_scan = 60 if (@time_to_scan -= 1) <= 0; return @time_to_scan > 0;`

Reply:
Haha I actually sat there staring at this for quite awhile when I first wrote it, trying to shorten it like that. Great minds, yea? I think in the end though, I ended up going with this for readability and simplicity. Anything I could do to shorten it wouldn't immediately be clear (to me at least) that it works. Your two-liner, for example looks like it'd always return true at first glance (I believe the condition would have to be`@time_to_scan == 60` instead. But then what if in the future there's some reason to manually set `@time_to_scan` to some value bigger than the max?)

Reply:
My example actually will work as written. The `-=` will return the final value of `@time_to_scan` because `foo -= bar` is syntactic sugar for `foo = foo - bar` and assignment returns the result of the assignment, so it'll evaluate to `foo`. Using that in conditions is actually a reasonably common idiom in ruby AFAIK.

Reply:
Oh I agree with that, but the `return @time_to_scan > 0;` is where I'm having trouble. If `time_to_scan` is `<= 0`, you've just set it back to 60, right? So there's no way that return ever won't be true (unless I'm reading it wrong?).

Reply:
`@time_to_scan = 60 && return false if (@time_to_scan -= 1) <= 0; return true`

Reply:
Ooh that's kinda nice. Looks like it needs to be `(time_to_scan = 60; return false)` in ruby for the one line if, but I'll go with it.

Reply:
`&&` also works because 60 is truthy, but yeah yours is neater.